### ISSUE TITLE
NullReferenceException in SyntaxInputNode.VisitTree

### ISSUE THREAD

**Issue by chsienki** [dotnet/roslyn#58647]
Add a new generator after a syntax tree has already been removed.

```
Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
Stack Trace:
   at System.Collections.Immutable.ImmutableArray`1.get_Length()
   at Microsoft.CodeAnalysis.SyntaxInputNode`1.Builder.VisitTree(Lazy`1 root, EntryState state, SemanticModel model, CancellationToken cancellationToken) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\Nodes\SyntaxInputNode.cs:line 79
   at Microsoft.CodeAnalysis.DriverStateTable.Builder.GetSyntaxInputTable(ISyntaxInputNode syntaxInputNode) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\Nodes\DriverStateTable.cs:line 104
   at Microsoft.CodeAnalysis.SyntaxInputNode`1.UpdateStateTable(Builder graphState, NodeStateTable`1 previousTable, CancellationToken cancellationToken) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\Nodes\SyntaxInputNode.cs:line 35
   at Microsoft.CodeAnalysis.DriverStateTable.Builder.GetLatestStateTableForNode[T](IIncrementalGeneratorNode`1 source) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\Nodes\DriverStateTable.cs:line 149
   at Microsoft.CodeAnalysis.SourceOutputNode`1.UpdateStateTable(Builder graphState, NodeStateTable`1 previousTable, CancellationToken cancellationToken) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\Nodes\SourceOutputNode.cs:line 41
   at Microsoft.CodeAnalysis.DriverStateTable.Builder.GetLatestStateTableForNode[T](IIncrementalGeneratorNode`1 source) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\Nodes\DriverStateTable.cs:line 149
   at Microsoft.CodeAnalysis.SourceOutputNode`1.AppendOutputs(IncrementalExecutionContext context, CancellationToken cancellationToken) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\Nodes\SourceOutputNode.cs:line 98
   at Microsoft.CodeAnalysis.GeneratorDriver.UpdateOutputs(ImmutableArray`1 outputNodes, IncrementalGeneratorOutputKind outputKind, Builder generatorRunStateBuilder, CancellationToken cancellationToken, Builder driverStateBuilder) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\GeneratorDriver.cs:line 283
   at Microsoft.CodeAnalysis.GeneratorDriver.RunGeneratorsCore(Compilation compilation, DiagnosticBag diagnosticsBag, CancellationToken cancellationToken) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\GeneratorDriver.cs:line 258
   at Microsoft.CodeAnalysis.GeneratorDriver.RunGenerators(Compilation compilation, CancellationToken cancellationToken) in C:\projects\roslyn\src\Compilers\Core\Portable\SourceGeneration\GeneratorDriver.cs:line 45
```

**Comment by jaredpar** [comment]
Can this be closed now that #58649 is merged?

**Comment by chsienki** [comment]
Fixed in  #58649
